lib/ansible/modules/cloud/podman/podman_container_info.py

```diff
@@ -0,0 +1,386 @@
+                    "NanoCpus": 0,
+                    "CgroupParent": "",
+                    "BlkioWeight": 0,
+                    "BlkioWeightDevice": null,
+                    "BlkioDeviceReadBps": null,
+                    "BlkioDeviceWriteBps": null,
+                    "BlkioDeviceReadIOps": null,
+                    "BlkioDeviceWriteIOps": null,
+                    "CpuPeriod": 0,
+                    "CpuQuota": 0,
+                    "CpuRealtimePeriod": 0,
+                    "CpuRealtimeRuntime": 0,
+                    "CpusetCpus": "",
+                    "CpusetMems": "",
+                    "Devices": [],
+                    "DiskQuota": 0,
+                    "KernelMemory": 0,
+                    "MemoryReservation": 0,
+                    "MemorySwap": 0,
+                    "MemorySwappiness": -1,
+                    "OomKillDisable": false,
+                    "PidsLimit": 0,
+                    "Ulimits": [
+                        {
+                            "Name": "RLIMIT_NOFILE",
+                            "Soft": 1048576,
+                            "Hard": 1048576
+                        },
+                        {
+                            "Name": "RLIMIT_NPROC",
+                            "Soft": 1048576,
+                            "Hard": 1048576
+                        }
+                    ],
+                    "CpuCount": 0,
+                    "CpuPercent": 0,
+                    "IOMaximumIOps": 0,
+                    "IOMaximumBandwidth": 0
+                }
+                }
+        ]
+"""
+
+import json
+from ansible.module_utils.basic import AnsibleModule
+
+
+def get_containers_facts(module, executable, name):
+    if not name:
+        all_names = [executable, 'container', 'ls', '-q', '-a']
+        rc, out, err = module.run_command(all_names)
+        if rc != 0:
+            module.fail_json(msg="Unable to get list of containers: %s" % err)
+        name = out.split()
+        if not name:
+            return [], out, err
+    command = [executable, 'container', 'inspect']
+    command.extend(name)
+    rc, out, err = module.run_command(command)
+    if rc == 0:
+        json_out = json.loads(out) if out else None
+        if json_out is None:
+            return [], out, err
+        return json_out, out, err
+    if rc != 0 and 'no such ' in err:
+        if len(name) < 2:
+            return [], out, err
+        return cycle_over(module, executable, name)
+    module.fail_json(msg="Unable to gather info for %s: %s" % (name or 'all containers', err))
+
+
+def cycle_over(module, executable, name):
+    inspection = []
+    for container in name:
+        command = [executable, 'container', 'inspect', container]
+        rc, out, err = module.run_command(command)
+        if rc == 0 and out:
+            json_out = json.loads(out)
+            if json_out:
+                inspection += json_out
+    return inspection, "", ""
```

Comment:
Why not return concatenated `err`? After all, you're returning this from the module.

Reply:
I think it will confuse users, there is no actually error if we collect data for existing modules only. I wouldn't like to bother people without a real reason.

Reply:
From other side they might be interested which containers failed to be inspected, so agree to add it.